src/Security/Http/Authentication/AuthenticationSuccessHandler.php

```diff
@@ -47,12 +51,33 @@ public function handleAuthenticationSuccess(UserInterface $user, $jwt = null): J
         $response              = new JsonResponse(['result' => 'ok', 'payload' => $this->responsePayload]);
         $response->headers->setCookie(new Cookie('BEARER', $data['token'], $decoded['exp'], '/', null, true, true, false, $this->sameSite));
 
-        return  $response;
+        return $response;
     }
 
     public function onAuthenticationSuccess(Request $request, TokenInterface $token): JsonResponse
     {
-        return $this->handleAuthenticationSuccess($token->getUser());
+        $response = $this->handleAuthenticationSuccess($token->getUser());
+
+        if ($this->rememberDeviceResolver->getRememberDeviceStatus()) {
+            if (is_null($request->cookies) || is_null($request->cookies->get('REMEMBER_DEVICE')) || $this->jwtEncoder->decode($request->cookies->get('REMEMBER_DEVICE'))['exp'] > time()) {
+                // Remove cookie if expired
+                $response->headers->removeCookie('REMEMBER_DEVICE');
+                // TODO: Add to InvalidToken table
+
```

Comment:
```suggestion
```
You can't remove AND add the same cookie in one response. This is not necessary either: the new cookie will replace any existing cookie with the same name